consensus/src/consensus_observer/observer/pending_blocks.rs

```diff
@@ -62,20 +65,27 @@ pub struct PendingBlockStore {
 
     // A map of ordered blocks that are without payloads. The key is
     // the (epoch, round) of the first block in the ordered block.
-    blocks_without_payloads: BTreeMap<(u64, Round), PendingBlockWithMetadata>,
+    blocks_without_payloads: BTreeMap<(u64, Round), Arc<PendingBlockWithMetadata>>,
+
+    // A map of ordered blocks that are without payloads. The key is
+    // the hash of the first block in the ordered block.
+    // Note: this is the same as blocks_without_payloads, but with a different key.
+    blocks_without_payloads_by_hash: BTreeMap<HashValue, Arc<PendingBlockWithMetadata>>,
```

Comment:
Two questions here:

1. Is this struct meant to emulate [`PendingBlocks`](https://github.com/aptos-labs/aptos-core/blob/main/consensus/src/block_storage/pending_blocks.rs#L13-L17) on the ConsensusObservor side? If so might be good to add a doc comment at the top of the struct referencing it

2. Does `blocks_without_payloads: BTreeMap<(u64, Round), Arc<PendingBlockWithMetadata>>` need both the `epoch` and `round` as keys? Why not just `round`?

Reply:
Also maybe `blocks_without_payloads` could be renamed to `blocks_without_payloads_by_epoch_round` or just `blocks_without_payloads_by_round` if epoch isn't necessary

Reply:
> Is this struct meant to emulate PendingBlocks on the ConsensusObservor side?

Hmm... I'm not sure. There seem to be some similarities, but I'm reluctant to say they represent exactly the same thing. Probably best to keep them separate. 🤔 

> Why not just round?

Consensus observer can buffer blocks across epoch changes (so we require both fields). This helps with performance around epoch transitions.

> Also maybe blocks_without_payloads could be renamed to blocks_without_payloads_by_epoch_round or just blocks_without_payloads_by_round if epoch isn't necessary

Hmm... the only reason to add `by_hash` was to disambiguate it from the primary store above. Might just leave it for now.